api/envoy/config/accesslog/v3/accesslog.proto

```diff
@@ -282,6 +286,19 @@ message GrpcStatusFilter {
   bool exclude = 2;
 }
 
+// Filters based on matching dynamic metadata.
+message MetadataFilter {
+  option (udpa.annotations.versioning).previous_message_type =
+      "envoy.config.filter.accesslog.v2.MetadataFilter";
+
+  // Matcher to check metadata for specified value. For example, to match on the access_log_hint metadata,
+  // set the filter to "envoy.common" and the path to "access_log_hint".
+  type.matcher.v3.MetadataMatcher matcher = 1;
+
+  // Default result if the key does not exist in dynamic metadata: if true, then log; if false, then don't log.
+  bool no_key_default = 2;
```

Comment:
What is logged in this case? Can you clarify the docs and link to the relevant default field?

I wonder which would be more common here in terms of default? Do you think the default should be true? I could go either way on this.

Reply:
That boolean field specifies whether to log if the dynamic metadata specified in the matcher doesn't exist, like in the case that there aren't any filters on the proxy that set the metadata. Istio logs everything by default, but there could be use cases for logging nothing by default.

I left the value of the field to be false if unset, but the more common use case might be to log everything unless otherwise specified.

Reply:
Yeah I think always logging by default is probably the way to go, or just leave this param out  for now and always log the default value inside the matcher? Either way can you clarify the docs here around what is logged if there is no match?

/wait

Reply:
To be clear if the key exists but is the wrong value, e.g. "access_log_hint" is false. Then the request will not be logged. If the key exists and matches then, the request will be logged. If the key does not exist, then we defer to the no_key_default field on whether to log or not. I'll clarify this in the docs and set the default value to be 'true'

Reply:
I forgot that in proto3 there isn't support for explicit defaults, so if I want no_key_default to be true by default, I would need to wrap it in a oneof. I can also remove this field and log by default. Or leave it false by default.

Reply:
You can use a Bool well known type. See other examples.

Reply:
Done. Thanks for the tip.